### PR TITLE
Update Storage

### LINKED ISSUE #34: Update storage to store contacts and appointments seperately
We might need to store contacts and appointments in 2 different files to make it easier to work with.

### LINKED ISSUE #59: Add `Appointment` to `ModelManager`
`ModelManager` only maintains `FilteredList<Person>` which allows for filtering of people in the GUI.

If we aim to get filters for appointments, we can extend `ModelManager` to handle `FilteredList<Appointment` as well. This would also require `AddressBook` to also wrap `appointments = new UniqueAppointmentList`.

New classes to create: `UniqueAppointmentList`.

Classes to update: `ModelManager`, `Addressbook`.

### PR DESCRIPTION
**Pull request by qinboan**
updated person storage to include UUID 
updated appointment storage

now json file would be written in this format
{persons[{details abt person A}, {details abt person B}], appointments[{details abt appt A}, {details abt appt B}]}
tldr an object containing 2 objects, persons and appointments, which each contains an arraylist of objects of that kind with all their details

resolves #34
resolves #59

### CODE REVIEW

**Comment by qinboan** [comment]
resolves #34 #59

File: src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateAppointmentsException();
+        }
+        internalList.add(toAdd);
+
+    }
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedAppointment}.
```
Comment (SampsonYe1999) [inline src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java:51]: Should be replacing the appointment in the list with editedAppointment?

Comment (qinboan) [reply src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java:51]: oh right i copied over the method from person LMAO, will fix it now

File: src/main/java/seedu/address/model/AddressBook.java [hunk 2]
```diff
@@ -94,12 +108,53 @@ public void removePerson(Person key) {
...
+     * Adds an appointment to the list of appointments.
+     *
+     * @param appointment The appointment to be added.
+     */
+    public void addAppointment(Appointment appointment) {
+        appointments.add(appointment);
+    }
+
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedPerson}.
```
Comment (SampsonYe1999) [inline src/main/java/seedu/address/model/AddressBook.java:133]: Need to be modified.

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -84,4 +86,38 @@ public interface Model {
...
+    void deleteAppointment(Appointment target);
+
+    /**
+     * Adds the given appointment.
+     * {@code person} must not already exist in the address book.
+     */
+    void addAppointment(Appointment appointment);
+
+    /**
+     * Replaces the given person {@code target} with {@code editedAppointment}.
```
Comment (SampsonYe1999) [inline src/main/java/seedu/address/model/Model.java:108]: Need to be modified.

**Review by SampsonYe1999 — DISMISSED** [review]
LGTM. Some mistakes in javaDocs are highlighted for modification.

**Review by SampsonYe1999 — APPROVED** [review]
LGTM 👍

**Comment by minreiseah** [comment]
This did not in fact update storage to store appointments. Refer to #71 for the implementation.
